src/app/templates/scripts/_server.js

```diff
@@ -1,26 +1,201 @@
-var connect = require('connect');
-var cowsay = require('cowsay');
-var path = require('path');
-var portscanner = require('portscanner');
-var serveStatic = require('serve-static');
+import Promise from 'bluebird';
+import browserify from 'browserify';
+import budo from 'budo';
+import fs from 'fs';
+import glob from 'glob';
+import mkdirp from 'mkdirp';
+<% if (sass) { -%>
+import sass from 'node-sass';
+<% } -%>
+import path from 'path';
+<% if (lang) { -%>
+import vjslangs from 'videojs-languages';
+<% } -%>
 
-// Configuration for the server.
-var PORT = 9999;
-var MAX_PORT = PORT + 100;
-var HOST = '127.0.0.1';
+/* eslint no-console: 0 */
 
-var app = connect();
+const pkg = require(path.join(__dirname, '../package.json'));
 
-app.use(serveStatic(path.join(__dirname, '..')));
+// Replace "%s" tokens with the plugin name in a string.
+const nameify = (str) =>
+  str.replace(/%s/g, pkg.name.split('/').reverse()[0]);
 
-portscanner.findAPortNotInUse(PORT, MAX_PORT, HOST, function(error, port) {
-  if (error) {
-    throw error;
-  }
+const srces = {
+<% if (sass) { -%>
+  css: 'src/plugin.scss',
+<% } -%>
+  js: 'src/plugin.js',
+<% if (lang) { -%>
+  langs: 'lang/*.json',
+<% } -%>
+  tests: glob.sync('test/**/*.test.js')
+};
+
+const dests = {
+<% if (sass) { -%>
+  css: nameify('dist/%s.css'),
+<% } -%>
+  js: nameify('dist/%s.js'),
+<% if (lang) { -%>
+  langs: 'dist/lang',
+<% } -%>
+  tests: 'test/dist/bundle.js'
+};
+
+const bundlers = {
 
-  process.stdout.write(cowsay.say({
-    text: 'Chewing the cud on ' + HOST + ':' + port
-  }) + '\n\n');
+  js: browserify({
+    debug: true,
+    entries: [srces.js],
+    standalone: nameify('%s'),
+    transform: [
+      'babelify',
+      'browserify-shim',
+      'browserify-versionify'
+    ]
+  }),
 
-  app.listen(port);
+  tests: browserify({
+    debug: true,
+    entries: srces.tests,
+    transform: [
+      'babelify',
+      'browserify-shim',
+      'browserify-versionify'
+    ]
+  })
+};
+
+const bundle = (name) =>
+  bundlers[name].bundle().pipe(fs.createWriteStream(dests[name]));
+
+const server = budo({
+  port: 9999,
+  stream: process.stdout
 });
+
+const reload = () => {
+  console.log('reloading');
+  server.reload();
+};
+
+/**
+ * A collection of functions which are mapped to strings that are used to
+ * generate RegExp objects. If a filepath matches the RegExp, the function
+ * will be used to handle that watched file.
+ *
+ * @type {Object}
+ */
+const handlers = {
```

Comment:
sounds like handler should be a `Map`. Or at least, we should make it one in a future update that drops support for older nodes.